x-pack/plugins/infra/public/containers/logs/with_log_flyout.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { Query } from 'react-apollo';
+import { FlyoutItemQuery, InfraLogItem } from '../../graphql/types';
+import { FlyoutVisibility } from '../../store/local/log_flyout';
+import { WithOptions } from '../with_options';
+import { flyoutItemQuery } from './flyout_item.gql_query';
+import { WithFlyoutOptions } from './with_log_flyout_options';
+
+interface WithFlyoutArgs {
+  flyoutItem: InfraLogItem | null;
+  isFlyoutVisible: boolean;
+  setFlyoutItem: (id: string) => void;
+  showFlyout: () => void;
+  hideFlyout: () => void;
+  error?: string | undefined;
+  loading: boolean;
+}
+
+interface WithFlyoutProps {
+  children: (args: WithFlyoutArgs) => React.ReactNode;
+}
+
+export const WithLogFlyout = ({ children }: WithFlyoutProps) => {
+  return (
+    <WithOptions>
+      {({ sourceId }) => (
+        <WithFlyoutOptions>
+          {({ showFlyout, hideFlyout, setFlyoutItem, flyoutId, flyoutVisibility }) => (
+            <Query<FlyoutItemQuery.Query, FlyoutItemQuery.Variables>
```

Comment:
The way these are nested means that requests are sent to the server whenever the `itemId` changes, even if the flyout is not visible. This happens even for the initial invalid empty `itemId`. How about wrapping this whole `<Query>` and its children into a conditional based on `flyoutVisibility` to prevent anything from being requested for a hidden flyout?

Reply:
Querying the empty `itemId` also produces long "ids can't be empty" stack traces in the Kibana server log.